pgxn/neon/pagestore_smgr.c

```diff
@@ -1061,8 +1061,7 @@ page_server_request(void const *req)
 	 * Current sharding model assumes that all metadata is present only at shard 0.
 	 * We still need to call get_shard_no() to check if shard map is up-to-date.
 	 */
-	if (((NeonRequest *) req)->tag != T_NeonGetPageRequest ||
-		((NeonGetPageRequest *) req)->forknum != MAIN_FORKNUM)
```

Comment:
No, we do not redirect requests between shards.
So if compute send request to wrong shard, it will cause error.

Actually this change removes inconsistency is shard calculation between PS and compute: in PS there is no special shard distribution for MAIN fork.

Moreover this second check is redundant, because `page_server_request` is never called for `get-age` request, so  first part of condition is always true. Should we completely remove this check? I am not sure.